File: 1to150/122_buyTimeToBuyandSellStcokII.java
/*
既然说明可以tansanction多次，那就意味着我可以今天卖今天再买。

所以只要今天的价格大于昨天的 我们就有获利，所以加上获利，最后得到的是最大获利
*/
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length == 0){
            return 0;
        }
        
        int profit = 0;
        
        for (int i = 1; i < prices.length; i++){
            if (prices[i] > prices[i - 1]){
                profit += prices[i] - prices[i - 1];
            }
        }
        
        return profit;
    }
}

File: 1to150/122_buyTimeToBuyandSellStcokII.py
'''
无限次交易， 只要今天的价格高于昨天的价格，我们就有获利， 所以running += pricc[i] - price[i - 1]

'''
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if not prices:
            return 0
        
        profit = 0
        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                profit += (prices[i] - prices[i - 1])
        
        return profit